### ISSUE TITLE
wesnoth.scenario.era may exist only on host

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#9586]
A check such as `wesnoth.scenario.era.id == 'xxx'` will cause an error if the other player does not have the era installed; the era tables does not exist then.
One can use  `wesnoth.scenario.mp_settings.era == 'xxx'` instead.

This should either be documented or changed.

**Comment by sevu** [comment]
Hoewever, `wesnoth.scenario.mp_settings.mp_era` is deprecated and recommends to use `wesnoth.scenario.era.id`.

**Comment by CelticMinstrel** [comment]
Hmm, that's kinda… annoying… I guess there's the option of synthesizing an era table from the mp_settings struct if the era isn't installed…?

**Comment by sevu** [comment]
I documented the limitation, closing this?

**Comment by gfgtdf** [comment]
No, having no way to figure out the era id on clinets which doesnt have the era installed is imo a problem. And we have to decide how to fix it (eg whetehr we should do what clemin suggested or whether we should un-deprecate mp_settings.mp_era or whether we shoudl add a new attribute)

My worry with celmins suggetion is that then there might not be a way for umc devs to figure out whether the era is installed or not which might lead umc to believe that the table returned by scenario.era.id is always the complete era which might lead ot other bugs.

Also the same problem might also apply to wesnoth.scenario.campaign

**Comment by ProditorMagnus** [comment]
Maybe wesnoth.scenario.resources and wesnoth.scenario.modifications are affected by same issue.
